src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/step_index_pattern/step_index_pattern.js

```diff
@@ -0,0 +1,205 @@
+    goToNextStep: PropTypes.func.isRequired,
+    initialQuery: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      matchingIndices: [],
+      isLoadingIndices: false,
+      query: props.initialQuery || '',
+      appendedWildcard: false,
+      showingIndexPatternQueryErrors: false,
+    };
+  }
+
+  fetchIndices = async (query) => {
+    const { esService } = this.props;
+
+    this.setState({ isLoadingIndices: true });
+    const matchingIndices = await getIndices(esService, `${query}*`);
+    createReasonableWait(() => this.setState({ matchingIndices, isLoadingIndices: false }));
+  }
+
+  onQueryChanged = (e) => {
+    const { appendedWildcard } = this.state;
+    const { target } = e;
+
+    let query = target.value;
+    if (query.length === 1 && canAppendWildcard(e.nativeEvent)) {
+      query += '*';
+      this.setState({ appendedWildcard: true });
+      setTimeout(() => target.setSelectionRange(1, 1));
+    }
+    else {
+      if (query === '*' && appendedWildcard) {
+        query = '';
+        this.setState({ appendedWildcard: false });
+      }
+    }
+
+    this.setState({ query, showingIndexPatternQueryErrors: !!query.length });
+    this.fetchIndices(query);
+  }
+
+  renderLoadingState() {
+    const { isLoadingIndices } = this.state;
+
+    if (isLoadingIndices) {
+      return (
+        <LoadingIndices/>
+      );
+    }
+
+    return null;
+  }
+
+  renderStatusMessage() {
+    const { initialIndices, isIncludingSystemIndices } = this.props;
+    const { query, isLoadingIndices, matchingIndices } = this.state;
+
+    if (isLoadingIndices) {
+      return null;
+    }
+
+    return (
+      <StatusMessage
+        initialIndices={initialIndices}
+        matchingIndices={matchingIndices}
+        query={query}
+        isIncludingSystemIndices={isIncludingSystemIndices}
+      />
+    );
+  }
+
+  renderList() {
+    const { isIncludingSystemIndices, initialIndices } = this.props;
+    const {
+      query,
+      matchingIndices,
+      isLoadingIndices,
+    } = this.state;
+
+    if (isLoadingIndices) {
+      return null;
+    }
+
+    const indices = getWhitelistedIndices(initialIndices, isIncludingSystemIndices, query, matchingIndices);
+    const indicesToList = query.length
+      ? indices.visibleIndices
+      : indices.initialIndices;
+
+    return (
+      <IndicesList
+        indices={indicesToList}
+      />
+    );
+  }
+
+  render() {
+    const { isIncludingSystemIndices, initialIndices, goToNextStep } = this.props;
+    const {
+      query,
+      showingIndexPatternQueryErrors,
+      matchingIndices,
+    } = this.state;
+
+    const {
+      exactMatchingIndices: indices
+    } = getWhitelistedIndices(initialIndices, isIncludingSystemIndices, query, matchingIndices);
+
+    let containsErrors = false;
+    const errors = [];
+    const characterList = ILLEGAL_CHARACTERS.slice(0, ILLEGAL_CHARACTERS.length - 1).join(', ');
+
```

Comment:
Yup, nice catch. I'm excluding the last one because it's an empty space and it looks off in the UI if I show that with a comma. Should I just show it? Or add a comment here?